pubsub/subscriptions/pull_concurrency.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_subscriber_concurrency_control]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func pullMsgsConcurrent(w io.Writer, projectID, subName string) error {
```

Comment:
Is this function tested?

Reply:
We currently don't have a good way to test this sample. 

https://github.com/GoogleCloudPlatform/golang-samples/blob/121fb3eec469dc207cc7ac3af7fdd7c394d862e5/pubsub/subscriptions/pull_concurrency.go#L41, `sub.Receive` is long running until the context is cancelled. I can refactor the sample to take in an external context, but I would probably need to add another comment like `ctx := context.Background` which might be confusing. What do you think?

Reply:
Ah. That's problematic...

Tests seem more important than the `context.Context` pattern. What do you think of passing a `context.Context` in or doing `ctx := context.WithTimeout(ctx, 5 * time.Second)` in the sample? I'm good with either, so whatever you prefer.